src/Pages/Cartpage/CartProduct.jsx

```diff
@@ -0,0 +1,178 @@
+        type: null,
+      });
+    }, 2000);
+  }, [showAlert.showAlert]);
+
+  const qtyIncrementHandler = () => {
+    updateCart({
+      product,
+      type: "increment",
+      token: authState.token,
+    });
+    setIsDisabled(true);
+    if (error) {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: error,
+        type: "danger",
+      });
+    } else {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: "Quantity Increased Successfully!",
+        type: "success",
+      });
+    }
+  };
+
+  const qtyDecrementHandler = () => {
+    updateCart({
+      product,
+      type: "decrement",
+      token: authState.token,
+    });
+    setIsDisabled(true);
+    if (error) {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: error,
+        type: "danger",
+      });
+    } else {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: "Quantity Decreased Successfully!",
+        type: "success",
+      });
+    }
+  };
+
+  const deleteCartItemHandler = () => {
+    removeFromCart({ product, token: authState.token });
+    if (error) {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: error,
+        type: "danger",
+      });
+    } else {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: "Item Deleted Successfully!",
+        type: "success",
+      });
+    }
+  };
+
+  const moveToWishlistHandler = () => {
+    if (!existsInWishlist) {
+      addToWishlist({
+        product,
+        token: authState.token,
+      });
+      deleteCartItemHandler();
+      if (error) {
+        setShowAlert({
+          showAlert: true,
+          alertMessage: error,
+          type: "danger",
+        });
+      } else {
+        setShowAlert({
+          showAlert: true,
+          alertMessage: "Product added to wishlist successfully!",
+          type: "success",
+        });
+        setIsDisabled(true);
+      }
+    } else {
+      setShowAlert({
+        showAlert: true,
+        alertMessage: "Product already exists in wishlist!",
+        type: "danger",
+      });
+    }
+  };
+
+  return (
+    <Card className="card-horizontal cart-card">
+      <div className="card__img">
+        <img src={image} alt={title} className="img-responsive" />
+      </div>
+      <div className="card__content">
+        <div className="card__header">
+          <h3 className="card__title">{title}</h3>
+          <p className="card__subtitle">{subtitle}</p>
+        </div>
+        <div className="card__amount">
+          <h4>₹{price}</h4>
+          <p className="strike-text small-text gray-text">
+            <del>₹{oldPrice}</del>
+          </p>
+          <p className="red-text discount">{discount}%</p>
+        </div>
+        <div className="card__rating">
+          <p>{rating.value}</p>
+          <FaStar />
+        </div>
+        <div className="card__quantity">
+          <span
+            className={`btn-icon ${
```

Comment:
For applying classnames this way. There's an npm library called **classNames**, it applies your classes based on condition. Maybe use that here.